### ISSUE TITLE
DateTime item-type not displaying consistently in classic UI

### ISSUE THREAD

**Issue by kaikreuzer** [eclipse/smarthome#614]
migrated from Bugzilla [#480822](https://bugs.eclipse.org/bugs/show_bug.cgi?id=480822)
status UNCONFIRMED severity *normal* in component *UI* for *---*
Reported in version *unspecified* on platform *PC*
Assigned to: Project Inbox

On 2015-10-27 20:57:43 -0400, Marcel Verpaalen wrote:

> I created a new binding for NTP in a OH2 way. However, having difficulties displaying the data.
> 
> When in the channel-type definition I define the channel with <item-type>DateTime</item-type>
> and do the update updateState(c.getUID(), new DateTimeType(calendar));
> It does not show consistently between the paper-UI and the classic UI.
> 
> If there is no <state>description, it displays fine in the paperUI, but it remains empty in the classic UI.
> 
> if I add <state readOnly="true" pattern="%1$tA, %1$td.%1$tm.%1$tY %1$tT"></state> it displays fine in classic UI, but in paperUI it shows {{getStateText(item)}}
> 
> Expected behavior is that even without specific pattern the DateTimeType can be displayed in classic UI.
> 
> Note: transfered from OH2 issue # 374

On 2015-10-28 13:55:31 -0400, Dennis Nobel wrote:

> > if I add <state readOnly="true" pattern="%1$tA, %1$td.%1$tm.%1$tY %1$tT"></state> it displays fine in classic UI, but in paperUI it shows {{getStateText(item)}}
> 
> As Sebastian already commented in the OH issue, the problem is, that the sprintf that is used in the Paper UI seems not to be compatible with the sprintf from Java.
> 
> > Expected behavior is that even without specific pattern the DateTimeType can be displayed in classic UI.
> 
> Usually you would define the pattern in the items.dsl or sitemap.dsl file I guess. There you can just override it for the Classic UI. But I donÂ´t know why the Classic UI does not have a fallback.

On 2015-10-28 14:07:00 -0400, Marcel Verpaalen wrote:

> 
> Indeed, seems like the 2 implementations of sprintf deviate.
> For now I'll remove the pattern from the ntp binding so it displays in paperUI, and than rely on the the pattern in the items.dsl or sitemap.dsl file to have the pattern to be used in classic UI.
> 
> However, still think indeed we need to find a way to have the fallback in classicUI as this will be an issue for other upcoming bindings as well

**Comment by htreu** [comment]
Since classic ui is out of maintenance I guess this can be closed @kaikreuzer?

**Comment by kaikreuzer** [comment]
No, Classic UI might not be in the focus, but it is still actively maintained.
